java-frontend/src/main/java/org/sonar/java/se/checks/OptionalGetBeforeIsPresentCheck.java

```diff
@@ -62,6 +66,21 @@ public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
     return visitor.programState;
   }
 
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    List<ProgramState> programStates = setNotPresentConstraint(context, syntaxNode);
+    Preconditions.checkState(programStates.size() == 1);
+    return programStates.get(0);
+  }
+
+  private static List<ProgramState> setNotPresentConstraint(CheckerContext context, Tree syntaxNode) {
+    SymbolicValue val = context.getState().peekValue();
+    if(syntaxNode.is(Tree.Kind.METHOD_INVOCATION) && OPTIONAL_EMPTY.matches(((MethodInvocationTree) syntaxNode))) {
+      return val.setConstraint(context.getState(), OptionalConstraint.NOT_PRESENT);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "val" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)